### PR TITLE
allow specifying timeout options

### PR DESCRIPTION
**Pull request by pnepywoda**
@markelliot @uschi2000

### CODE REVIEW

File: http-clients/src/main/java/com/palantir/remoting/http/FeignClients.java [hunk 1]
```diff
@@ -41,6 +45,14 @@ private FeignClients() {}
...
      */
     public static FeignClientFactory standard() {
+        return standard(DEFAULT_TIMEOUT_OPTIONS);
+    }
+
+    /**
+     * Provides a {@link FeignClientFactory} with an {@link ObjectMapper} configured with {@link
+     * com.fasterxml.jackson.datatype.guava.GuavaModule} and {@link com.fasterxml.jackson.datatype.jdk7.Jdk7Module}.
+     */
+    public static FeignClientFactory standard(Request.Options timeoutOptions) {
```
Comment (uschi2000) [inline http-clients/src/main/java/com/palantir/remoting/http/FeignClients.java:55]: seems like we need a BuilderFactoryBuilder :(

**Comment by uschi2000** [comment]
looks OK

File: http-clients/src/main/java/com/palantir/remoting/http/FeignClients.java (hunk 1 above)
Comment (pnepywoda) [inline http-clients/src/main/java/com/palantir/remoting/http/FeignClients.java:55]: > Yuck

.....or call it Warehouse instead ;)

**Comment by markelliot** [comment]
What if we did FeignClients.withTimeout(...).standard()?

Could keep things internal to FeignClients?

Am worried at this point we're just recreating the Feign builder and ought to pick a different wrapping point.

Sent from my iPhone

On Feb 25, 2016, at 7:48 AM, Robert Fink <notifications@github.com<mailto:notifications@github.com>> wrote:

looks OK

File: http-clients/src/main/java/com/palantir/remoting/http/FeignClients.java (hunk 1 above)
Comment (jlzhao27) [inline http-clients/src/main/java/com/palantir/remoting/http/FeignClients.java:55]: this only sets read and connection right, what if we want write timeout?

Comment (uschi2000) [inline http-clients/src/main/java/com/palantir/remoting/http/FeignClients.java:55]: there's currently no good support for write timeout in feign. should
probably fix it there first.

Comment (pnepywoda) [inline http-clients/src/main/java/com/palantir/remoting/http/FeignClients.java:55]: Ok yes many things can be improved in the future, but this is specifically
so Foundry can stop copy/pasting the method with timeoutOptions and simply
use http-remoting. Are there objections to this commit as-is?

Comment (markelliot) [inline http-clients/src/main/java/com/palantir/remoting/http/FeignClients.java:55]: It's probably fine, we need to make some real investment in this library in the near future.

Sent from my iPhone

Comment (pnepywoda) [inline http-clients/src/main/java/com/palantir/remoting/http/FeignClients.java:55]: Ok. I'll wait patiently for a +1 ;)

**Pull request merged**